scripts/hostmap-crtsh.nse

```diff
@@ -88,31 +61,51 @@ local function is_valid_hostname (name)
   return true
 end
 
+-- TRUE subdomain check (ends with .parent)
+local function is_subdomain(name, parent)
+  return name:match("%." .. parent .. "$") ~= nil
```

Comment:
There are a few deficiencies here:
* Some names embedded in certificates include upper-case characters, so you should be using case-insensitive comparison.
* The search term is not escaping special characters in `parent`, so it is prone to both false positives and negatives.
* There is no need to use `~= nil`. The code would work the same without it.
* Function `string.find` is slightly faster than `string.match` due not having to allocate a new string (as the output).

Overall, you might be better off to use straight string comparison, instead of a search term.